src/transaction/txParser.ts

```diff
@@ -283,14 +342,16 @@ const parseMetaDataHash = (metaDataHash: any): Buffer | null => {
   return metaDataHash || null
 }
 
+const parseTxMint = (mint: any): _Mint => parseMultiAsset(mint, true)
+
 const deconstructUnsignedTxDecoded = (unsignedTxDecoded: any): _UnsignedTxDecoded => {
-  if (unsignedTxDecoded.length === 2) return unsignedTxDecoded
+  if (unsignedTxDecoded.length === 2) {
+    const [txBody, meta] = unsignedTxDecoded
+    return { txBody, meta, scriptWitnesses: [] }
+  }
   if (unsignedTxDecoded.length === 3) {
-    const [txBody, scriptWitnesses, metaData] = unsignedTxDecoded
-    if (scriptWitnesses != null && scriptWitnesses?.length > 0) {
-      throw Error(Errors.ScriptWitnessesNotSupported)
-    }
-    return [txBody, metaData]
+    const [txBody, scriptWitnesses, meta] = unsignedTxDecoded
+    return { txBody, scriptWitnesses, meta }
```

Comment:
Is it actually expected for `scriptWitnesses` to come in cbor as an array? I was under impression that they should be in object under key `1` as defined in https://github.com/input-output-hk/cardano-ledger-specs/blob/master/alonzo/test/cddl-files/alonzo.cddl#L253

In the test you added it's visible that `scriptWitnesses` are just an array in `unsignedCborHex`, but in `signedTxCborHex` they are in object under key `1`:
https://github.com/vacuumlabs/cardano-hw-cli/blob/ja/multisig/test/integration/ledger/node/tx.js#L175-L181

Which also brings a question, how did you create unsigned tx cbor, which tool are we integrating to? Was it done with cardano-cli? If so can you give me some directions?

Reply:
I sent you directions on how to create the transaction via cardano-cli offline. Raw transactions built by cardano-cli contain the scripts as an array. Signed transaction however correctly contain the scripts in the witnesses under the key 1. So I think the code here should be correct.

Reply:
Do I understand that cardano-cli is doing something wrong? I don't think we should make our code not aligned with specs just because cardano-cli does not follow them --- we actually do not know how the users of our hw-cli create the transactions. Or at least we should have a clear understanding of what cardano-cli is doing and why --- won't it change with Alonzo, for instance? (I bet it will since there is tons of new stuff in transaction_witness_set.) Please ask IOHK and badger them until they respond properly, perhaps @gabrielKerekes ?

Reply:
No. They include the script witnesses as an array in the **raw (unsigned)** transaction so they can later be simply parsed out during transaction signing and included in the signed transaction according to the spec. I think it's reasonable for HW CLI to do the same.